crates/burn-core/src/nn/prelu.rs

```diff
@@ -0,0 +1,47 @@
+use crate as burn;
+use crate::config::Config;
+use crate::module::Module;
+use crate::module::Param;
+use crate::nn::Initializer;
+use crate::tensor::backend::Backend;
+use crate::tensor::Tensor;
+/// Parametric Relu layer.
+#[derive(Module, Debug)]
+pub struct PRelu<B: Backend> {
+    alpha: Param<Tensor<B, 1>>,
+    num_parameters: usize,
```

Comment:
I don't think we need the `num_parameters` field. Also I think we should put the `alpha` parameter public and document it, I think it's a better API to make the parameters of module public.

Reply:
Why is the `num_parameters` field not required. The number of weight can be either 1 (same weight across all channels) or `num_channel` (different weight for each channel which needs to be learnt)

Reply:
Referring to the docs from pytorch [here](https://pytorch.org/docs/stable/generated/torch.nn.PReLU.html)
and Candle implementation [here](https://github.com/huggingface/candle/blob/45d5322d62d59a2c9be5fe8c642d0fa56fbb73b1/candle-nn/src/activation.rs#L46C1-L49C2)

Reply:
It's never read, the information is inside the alpha shape, so no need to have another field.

Reply:
ok